### PR TITLE
Enable paste in textboxes

### PR DESCRIPTION
**Pull request by DemonWav**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -361,22 +366,30 @@ fn handle_window_event(window: &sdl2::video::Window,
...
             if game.focused {
                 if let Some(steven_key) = settings::Stevenkey::get_by_keycode(key, &game.vars) {
                     game.server.key_press(true, steven_key);
                 }
             } else {
-                ui_container.key_press(game, key, true);
+                let mut ctrl_pressed = false;
+                if keymod.intersects(keyboard::LCTRLMOD | keyboard::RCTRLMOD) {
+                    ctrl_pressed = true;
+                }
```
Comment (Thinkofname) [inline src/main.rs]: this could be `let ctrl_pressed = keymod.intersects(keyboard::LCTRLMOD | keyboard::RCTRLMOD)` (and below too)

Comment (DemonWav) [inline src/main.rs]: Right. Oops.

File: src/ui/mod.rs [hunk 2]
```diff
@@ -1321,6 +1321,17 @@ impl UIElement for TextBox {
...
                 self.submit_funcs.append(&mut temp);
             },
+            (Keycode::V, true) => {
+                if ctrl_pressed {
+                    let clipboard = game.sdl.video().unwrap().clipboard();
+                    if clipboard.has_clipboard_text() {
+                        let text = clipboard.clipboard_text().unwrap();
+                        for c in text.chars() {
+                            self.input.push(c);
+                        }
```
Comment (Thinkofname) [inline src/ui/mod.rs]: I think you could just use `self.input.push_str(&text)` instead here

**Comment by Thinkofname** [comment]
Thanks

**Pull request merged**